informalsystems/ibc-rs
# Add telemetry and Prometheus endpoint

Closes: #868

## Description

Adds telemetry to Hermes and exposes an Prometheus endpoint.

## Testing

* Configure telemetry exporter in config.toml by adding a `[telemetry]` section:
```
[telemetry]
enabled = true
port = 3001
```

1. Start dev environment, for example with 3 chains in a config.toml:
```
./scripts/dev-env ~/.hermes/config.toml ibc-0 ibc-1
```

2. Run Hermes:
```
$ cargo run -- start
```

In a separate terminal, watch the telemetry (please ensure the port matches the one specified in the `telemetry.port` config parameter
```
$ watch -n 1 curl -s localhost:3001/metrics
```

Expect the output to look like
```
# HELP hermes_chains_num Number of chains the relay is connecting to
# TYPE hermes_chains_num counter
hermes_chains_num{hermes="all"} 2
# HELP hermes_channels_num Number of channels the relay is connecting to
# TYPE hermes_channels_num counter
hermes_channels_num{hermes="all"} 2
# HELP hermes_tx_msg_ibc_acknowledge_packet Total number of IBC packets acknowledged
# TYPE hermes_tx_msg_ibc_acknowledge_packet counter
hermes_tx_msg_ibc_acknowledge_packet{hermes="all"} 2
```

Run the e2e test trigger other metrics....
______

For contributor use:

- [ ] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
